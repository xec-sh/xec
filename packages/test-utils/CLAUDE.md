# @xec-sh/test-utils - Shared Testing Utilities

## 🎯 Package Mission
Provides shared testing utilities, Docker containers, and Kubernetes helpers for consistent testing across the Xec monorepo.

## 🏛 Architecture

```
├── docker/           # Docker test utilities
│   ├── container-manager.ts  # Container lifecycle management
│   └── utils.ts             # Docker API utilities
├── k8s/              # Kubernetes test utilities
│   └── kind-cluster-manager.ts  # Kind cluster management
├── helpers/          # General test helpers
│   ├── ssh-test-helpers.ts    # SSH testing utilities
│   └── docker-ssh-manager.sh  # Shell script for containers
└── index.ts          # Public exports
```

## 🐳 Docker Test Containers

### Available Containers
- `ubuntu-apt` (port 2201) - Ubuntu with APT
- `centos7-yum` (port 2202) - CentOS 7 with YUM
- `fedora-dnf` (port 2203) - Fedora with DNF
- `alpine-apk` (port 2204) - Alpine with APK
- `manjaro-pacman` (port 2205) - Manjaro with Pacman
- `ubuntu-brew` (port 2206) - Ubuntu with Homebrew
- `ubuntu-snap` (port 2207) - Ubuntu with Snap

### Container Management
```typescript
import { dockerManager } from '@xec-sh/test-utils';

// Start specific container
await dockerManager.startContainer('ubuntu-apt');

// Start all containers
await dockerManager.startAllContainers();

// Check if Docker is available
if (dockerManager.isDockerAvailable()) {
  // Run Docker tests
}
```

## 🚢 Kubernetes Testing

### Kind Cluster Management
```typescript
import { KindClusterManager } from '@xec-sh/test-utils';

const cluster = new KindClusterManager({ name: 'test-cluster' });

// Create cluster
await cluster.createCluster();

// Deploy test pod
await cluster.deployTestPod('test-pod', 'test-namespace');

// Run kubectl commands
const pods = cluster.kubectl('get pods -n test');

// Cleanup
await cluster.deleteCluster();
cluster.cleanup();
```

## 🧪 SSH Test Helpers

### Test Suite Wrapper
```typescript
import { describeSSH, getSSHConfig, testEachPackageManager } from '@xec-sh/test-utils';

// Automatically manages Docker containers
describeSSH('SSH Tests', () => {
  testEachPackageManager('should install package', async (container) => {
    const config = getSSHConfig(container.name);
    // Test implementation
  });
});
```

### Configuration
```typescript
interface SSHTestConfig {
  containers?: string[];      // Specific containers to start
  timeout?: number;          // Container operation timeout
  skipIfNoDocker?: boolean;  // Skip if Docker unavailable
}
```

## 🔧 Docker Utilities

### Container Information
```typescript
import { getContainerInfo, waitForContainer } from '@xec-sh/test-utils';

const info = await getContainerInfo('mycontainer');
await waitForContainer('mycontainer', 5000);
```

### Container Execution
```typescript
import { execInContainer } from '@xec-sh/test-utils';

const result = await execInContainer('mycontainer', ['ls', '-la']);
console.log(result.stdout);
```

### Container Logs
```typescript
import { getContainerLogs } from '@xec-sh/test-utils';

const { stdout, stderr } = await getContainerLogs('mycontainer');
```

## 📋 Usage in Tests

### In package.json
```json
{
  "devDependencies": {
    "@xec-sh/test-utils": "workspace:*"
  }
}
```

### In Tests
```typescript
import { 
  dockerManager,
  KindClusterManager,
  describeSSH,
  SSH_TEST_CONFIGS 
} from '@xec-sh/test-utils';

// Use utilities as needed
```

## 🚨 Important Notes

1. **Docker Dependency**: Most utilities require Docker to be installed
2. **Port Conflicts**: Ensure ports 2201-2207 are available for SSH containers
3. **Cleanup**: Containers are automatically cleaned up after tests
4. **CI Optimization**: Minimal container set used in CI environments
5. **Symlink Structure**: Docker images are symlinked from root `/docker` directory

## ⚡ Performance Tips

- Use `startRequiredContainers()` for faster startup
- Containers are reused between test runs when possible
- Kind clusters persist during test session
- SSH connections are pooled and reused

## 🔮 Future Enhancements

- Additional container images (more distros/package managers)
- Podman support alongside Docker
- Test fixture generation utilities
- Mock service containers (databases, message queues)
- Performance benchmarking utilities