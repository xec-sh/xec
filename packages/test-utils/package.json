{
  "name": "@xec-sh/test-utils",
  "version": "0.7.0",
  "description": "Shared test utilities",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "clean": "rm -rf dist",
    "test": "jest --passWithNoTests",
    "typecheck": "tsc --noEmit"
  },
  "keywords": [
    "xec",
    "test",
    "utils",
    "docker",
    "kubernetes"
  ],
  "author": "LuxQuant",
  "license": "MIT",
  "dependencies": {
    "dockerode": "4.0.7"
  },
  "devDependencies": {
    "@types/dockerode": "3.3.42",
    "@types/jest": "30.0.0",
    "@types/node": "24.1.0",
    "jest": "30.0.5",
    "ts-jest": "29.4.0",
    "typescript": "5.8.3"
  },
  "files": [
    "dist",
    "docker"
  ]
}
