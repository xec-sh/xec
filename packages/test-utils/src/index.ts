// Docker utilities
export * from './docker/container-manager';
// Kubernetes utilities
export * from './k8s/kind-cluster-manager';

// SSH test helpers
export * from './helpers/ssh-test-helpers';

export { 
  docker,
  execInContainer,
  getContainerInfo,
  getContainerLogs,
  type ContainerInfo,
  cleanupTestContainers
} from './docker/utils';

// SSH connection configuration
export interface SSHConnectionConfig {
  host: string;
  port: number;
  username: string;
  password: string;
  packageManager: string;
  testPackage: string;
}

// Re-export container configurations with SSH details
export const SSH_TEST_CONFIGS: SSHConnectionConfig[] = [
  {
    host: 'localhost',
    port: 2201,
    username: 'user',
    password: 'password',
    packageManager: 'apt',
    testPackage: 'curl'
  },
  {
    host: 'localhost',
    port: 2202,
    username: 'user',
    password: 'password',
    packageManager: 'yum',
    testPackage: 'wget'
  },
  {
    host: 'localhost',
    port: 2203,
    username: 'user',
    password: 'password',
    packageManager: 'dnf',
    testPackage: 'nano'
  },
  {
    host: 'localhost',
    port: 2204,
    username: 'user',
    password: 'password',
    packageManager: 'apk',
    testPackage: 'vim'
  },
  {
    host: 'localhost',
    port: 2205,
    username: 'user',
    password: 'password',
    packageManager: 'pacman',
    testPackage: 'htop'
  },
  {
    host: 'localhost',
    port: 2206,
    username: 'user',
    password: 'password',
    packageManager: 'brew',
    testPackage: 'jq'
  },
  {
    host: 'localhost',
    port: 2207,
    username: 'user',
    password: 'password',
    packageManager: 'snap',
    testPackage: 'hello'
  }
];

// Helper to get SSH config by container name
export function getSSHConfigByName(containerName: string): SSHConnectionConfig | undefined {
  const containerMap: Record<string, number> = {
    'ubuntu-apt': 2201,
    'centos7-yum': 2202,
    'fedora-dnf': 2203,
    'alpine-apk': 2204,
    'manjaro-pacman': 2205,
    'ubuntu-brew': 2206,
    'ubuntu-snap': 2207
  };

  const port = containerMap[containerName];
  return SSH_TEST_CONFIGS.find(config => config.port === port);
}