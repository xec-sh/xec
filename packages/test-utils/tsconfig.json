{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declaration": true,
    "declarationMap": true,
    "target": "ES2020",
    "module": "commonjs",
    "moduleResolution": "node"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "**/*.test.ts"]
}