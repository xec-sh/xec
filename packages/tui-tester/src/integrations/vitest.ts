/**
 * Vitest Integration
 * Custom matchers and utilities for Vitest testing framework
 */

import { expect } from 'vitest';

import { normalizeText } from '../core/utils.js';
import { SnapshotManager } from '../snapshot/snapshot-manager.js';

import type { ScreenCapture } from '../core/types.js';

declare module 'vitest' {
  interface Assertion<T = any> {
    toMatchTerminalSnapshot(name?: string): Promise<void>;
    toContainText(text: string): void;
    toMatchPattern(pattern: RegExp): void;
    toHaveSize(cols: number, rows: number): void;
  }
  
  interface AsymmetricMatchersContaining {
    terminalSnapshot(name?: string): any;
    containingText(text: string): any;
  }
}

/**
 * Setup Vitest matchers for terminal testing
 */
export function setupVitestMatchers(snapshotManager?: SnapshotManager) {
  const manager = snapshotManager || new SnapshotManager({
    updateSnapshots: process.env.UPDATE_SNAPSHOTS === 'true',
    snapshotDir: './__snapshots__'
  });

  // Add custom matchers
  expect.extend({
    /**
     * Match terminal snapshot
     */
    async toMatchTerminalSnapshot(
      received: ScreenCapture,
      name?: string
    ) {
      const testPath = (this as any).testPath;
      const testName = (this as any).currentTestName;
      const snapshotName = name || testName || 'terminal-snapshot';
      
      const result = await manager.matchSnapshot(received, snapshotName, testPath);
      
      return {
        pass: result.pass,
        message: () => result.message || result.diff || 'Snapshot comparison failed',
        actual: received.text,
        expected: result.pass ? received.text : undefined
      };
    },

    /**
     * Check if capture contains text
     */
    toContainText(received: ScreenCapture, text: string) {
      const normalized = normalizeText(received.text, { ignoreCase: false });
      const searchText = normalizeText(text, { ignoreCase: false });
      const pass = normalized.includes(searchText);
      
      return {
        pass,
        message: () => 
          pass 
            ? `Expected screen not to contain "${text}"`
            : `Expected screen to contain "${text}"\n\nScreen content:\n${received.text}`
      };
    },

    /**
     * Match pattern in screen content
     */
    toMatchPattern(received: ScreenCapture, pattern: RegExp) {
      const pass = pattern.test(received.text);
      
      return {
        pass,
        message: () =>
          pass
            ? `Expected screen not to match pattern ${pattern}`
            : `Expected screen to match pattern ${pattern}\n\nScreen content:\n${received.text}`
      };
    },

    /**
     * Check screen size
     */
    toHaveSize(received: ScreenCapture, cols: number, rows: number) {
      const pass = received.size.cols === cols && received.size.rows === rows;
      
      return {
        pass,
        message: () =>
          pass
            ? `Expected screen not to have size ${cols}x${rows}`
            : `Expected screen to have size ${cols}x${rows}, but got ${received.size.cols}x${received.size.rows}`
      };
    }
  });

  return manager;
}