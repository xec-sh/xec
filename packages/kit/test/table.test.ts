import { vi, test, expect, afterAll, describe, afterEach, beforeAll, beforeEach } from 'vitest';

import * as prompts from '../src/index.js';
import { prism as colors } from '../src/index.js';
import { MockReadable, MockWritable } from './test-utils.js';

describe.each(['true', 'false'])('table (isCI = %s)', (isCI) => {
  let originalCI: string | undefined;
  let output: MockWritable;
  let input: MockReadable;

  beforeAll(() => {
    originalCI = process.env['CI'];
    process.env['CI'] = isCI;
  });

  afterAll(() => {
    process.env['CI'] = originalCI;
  });

  beforeEach(() => {
    output = new MockWritable();
    input = new MockReadable();
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  test('renders basic table with data', () => {
    const data = [
      { id: 1, name: 'Alice', age: 30 },
      { id: 2, name: 'Bob', age: 25 },
      { id: 3, name: 'Charlie', age: 35 },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'id', header: 'ID', width: 5 },
        { key: 'name', header: 'Name', width: 15 },
        { key: 'age', header: 'Age', width: 5, align: 'right' },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with single border style', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      borders: 'single',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with double border style', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      borders: 'double',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with rounded border style', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      borders: 'rounded',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with ascii border style', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      borders: 'ascii',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with no borders', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      borders: 'none',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders empty table', () => {
    prompts.table({
      data: [],
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with custom formatters', () => {
    const data = [
      { name: 'Alice', score: 95, status: true },
      { name: 'Bob', score: 78, status: false },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        {
          key: 'score',
          header: 'Score',
          width: 10,
          format: (v) => `${v}%`,
          align: 'right',
        },
        {
          key: 'status',
          header: 'Status',
          width: 10,
          format: (v) => (v ? '✓' : '✗'),
        },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with custom styles', () => {
    const data = [
      { name: 'Alice', status: 'active' },
      { name: 'Bob', status: 'inactive' },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        {
          key: 'status',
          header: 'Status',
          width: 10,
          style: (text, value) => (value === 'active' ? colors.green(text) : colors.red(text)),
        },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders table with text alignment', () => {
    const data = [{ left: 'L', center: 'C', right: 'R' }];

    prompts.table({
      data,
      columns: [
        { key: 'left', header: 'Left', width: 10, align: 'left' },
        { key: 'center', header: 'Center', width: 10, align: 'center' },
        { key: 'right', header: 'Right', width: 10, align: 'right' },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('truncates long text with ellipsis', () => {
    const data = [{ text: 'This is a very long text that should be truncated' }];

    prompts.table({
      data,
      columns: [{ key: 'text', header: 'Text', width: 15, ellipsis: true }],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('handles wide characters (CJK)', () => {
    const data = [
      { name: '太郎', description: '日本語のテスト' },
      { name: '花子', description: 'もう一つのテスト' },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: '名前', width: 10 },
        { key: 'description', header: '説明', width: 20 },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders with header styling', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      headerStyle: colors.cyan,
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders without header', () => {
    const data = [{ name: 'Test', value: '123' }];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      showHeader: false,
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('handles null and undefined values', () => {
    const data = [
      { name: 'Alice', value: null },
      { name: null, value: 'Test' },
      { name: 'Bob', value: undefined },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 10 },
        { key: 'value', header: 'Value', width: 10 },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders with auto width', () => {
    const data = [
      { short: 'A', long: 'This is a longer text' },
      { short: 'B', long: 'Short' },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'short', header: 'Short', width: 'auto' },
        { key: 'long', header: 'Long', width: 'auto' },
      ],
      width: 'auto',
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('renders with content width', () => {
    const data = [
      { name: 'A', description: 'First' },
      { name: 'BB', description: 'Second item' },
      { name: 'CCC', description: 'Third' },
    ];

    prompts.table({
      data,
      columns: [
        { key: 'name', header: 'Name', width: 'content' },
        { key: 'description', header: 'Description', width: 'content' },
      ],
      output,
    });

    expect(output.buffer).toMatchSnapshot();
  });

  test('throws error for invalid data', () => {
    expect(() => {
      prompts.table({
        data: null as any,
        columns: [{ key: 'name', header: 'Name' }],
        output,
      });
    }).toThrow('Table data must be an array');
  });

  test('throws error for empty columns', () => {
    expect(() => {
      prompts.table({
        data: [],
        columns: [],
        output,
      });
    }).toThrow('Table must have at least one column');
  });
});
