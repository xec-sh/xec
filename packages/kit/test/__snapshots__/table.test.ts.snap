// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`table (isCI = false) > handles null and undefined values 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Alice      │            │
│            │ Test       │
│ Bob        │            │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > handles wide characters (CJK) 1`] = `
[
  "┌────────────┬──────────────────────┐
│ [1m名前      [22m │ [1m説明                [22m │
├────────────┼──────────────────────┤
│ 太郎       │ 日本語のテスト       │
│ 花子       │ もう一つのテスト     │
└────────────┴──────────────────────┘
",
]
`;

exports[`table (isCI = false) > renders basic table with data 1`] = `
[
  "┌───────┬─────────────────┬───────┐
│ [1mID   [22m │ [1mName           [22m │ [1m  Age[22m │
├───────┼─────────────────┼───────┤
│ 1     │ Alice           │    30 │
│ 2     │ Bob             │    25 │
│ 3     │ Charlie         │    35 │
└───────┴─────────────────┴───────┘
",
]
`;

exports[`table (isCI = false) > renders empty table 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
[2m│        (no data)        │[22m
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders table with ascii border style 1`] = `
[
  "+------------+------------+
| [1mName      [22m | [1mValue     [22m |
+------------+------------+
| Test       | 123        |
+------------+------------+
",
]
`;

exports[`table (isCI = false) > renders table with custom formatters 1`] = `
[
  "┌────────────┬────────────┬────────────┐
│ [1mName      [22m │ [1m     Score[22m │ [1mStatus    [22m │
├────────────┼────────────┼────────────┤
│ Alice      │        95% │ ✓          │
│ Bob        │        78% │ ✗          │
└────────────┴────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders table with custom styles 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mStatus    [22m │
├────────────┼────────────┤
│ Alice      │ [32mactive    [39m │
│ Bob        │ [31minactive  [39m │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders table with double border style 1`] = `
[
  "╔════════════╦════════════╗
║ [1mName      [22m ║ [1mValue     [22m ║
╠════════════╬════════════╣
║ Test       ║ 123        ║
╚════════════╩════════════╝
",
]
`;

exports[`table (isCI = false) > renders table with no borders 1`] = `
[
  "[1mName      [22m [1mValue     [22m
Test       123       
",
]
`;

exports[`table (isCI = false) > renders table with rounded border style 1`] = `
[
  "╭────────────┬────────────╮
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Test       │ 123        │
╰────────────┴────────────╯
",
]
`;

exports[`table (isCI = false) > renders table with single border style 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders table with text alignment 1`] = `
[
  "┌────────────┬────────────┬────────────┐
│ [1mLeft      [22m │ [1m  Center  [22m │ [1m     Right[22m │
├────────────┼────────────┼────────────┤
│ L          │     C      │          R │
└────────────┴────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders with auto width 1`] = `
[
  "┌───────┬───────────────────────┐
│ [1mShort[22m │ [1mLong                 [22m │
├───────┼───────────────────────┤
│ A     │ This is a longer text │
│ B     │ Short                 │
└───────┴───────────────────────┘
",
]
`;

exports[`table (isCI = false) > renders with content width 1`] = `
[
  "┌──────┬─────────────┐
│ [1mName[22m │ [1mDescription[22m │
├──────┼─────────────┤
│ A    │ First       │
│ BB   │ Second item │
│ CCC  │ Third       │
└──────┴─────────────┘
",
]
`;

exports[`table (isCI = false) > renders with header styling 1`] = `
[
  "┌────────────┬────────────┐
│ [36mName      [39m │ [36mValue     [39m │
├────────────┼────────────┤
│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > renders without header 1`] = `
[
  "│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = false) > truncates long text with ellipsis 1`] = `
[
  "┌─────────────────┐
│ [1mText           [22m │
├─────────────────┤
│ This is a ve... │
└─────────────────┘
",
]
`;

exports[`table (isCI = true) > handles null and undefined values 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Alice      │            │
│            │ Test       │
│ Bob        │            │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > handles wide characters (CJK) 1`] = `
[
  "┌────────────┬──────────────────────┐
│ [1m名前      [22m │ [1m説明                [22m │
├────────────┼──────────────────────┤
│ 太郎       │ 日本語のテスト       │
│ 花子       │ もう一つのテスト     │
└────────────┴──────────────────────┘
",
]
`;

exports[`table (isCI = true) > renders basic table with data 1`] = `
[
  "┌───────┬─────────────────┬───────┐
│ [1mID   [22m │ [1mName           [22m │ [1m  Age[22m │
├───────┼─────────────────┼───────┤
│ 1     │ Alice           │    30 │
│ 2     │ Bob             │    25 │
│ 3     │ Charlie         │    35 │
└───────┴─────────────────┴───────┘
",
]
`;

exports[`table (isCI = true) > renders empty table 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
[2m│        (no data)        │[22m
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders table with ascii border style 1`] = `
[
  "+------------+------------+
| [1mName      [22m | [1mValue     [22m |
+------------+------------+
| Test       | 123        |
+------------+------------+
",
]
`;

exports[`table (isCI = true) > renders table with custom formatters 1`] = `
[
  "┌────────────┬────────────┬────────────┐
│ [1mName      [22m │ [1m     Score[22m │ [1mStatus    [22m │
├────────────┼────────────┼────────────┤
│ Alice      │        95% │ ✓          │
│ Bob        │        78% │ ✗          │
└────────────┴────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders table with custom styles 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mStatus    [22m │
├────────────┼────────────┤
│ Alice      │ [32mactive    [39m │
│ Bob        │ [31minactive  [39m │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders table with double border style 1`] = `
[
  "╔════════════╦════════════╗
║ [1mName      [22m ║ [1mValue     [22m ║
╠════════════╬════════════╣
║ Test       ║ 123        ║
╚════════════╩════════════╝
",
]
`;

exports[`table (isCI = true) > renders table with no borders 1`] = `
[
  "[1mName      [22m [1mValue     [22m
Test       123       
",
]
`;

exports[`table (isCI = true) > renders table with rounded border style 1`] = `
[
  "╭────────────┬────────────╮
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Test       │ 123        │
╰────────────┴────────────╯
",
]
`;

exports[`table (isCI = true) > renders table with single border style 1`] = `
[
  "┌────────────┬────────────┐
│ [1mName      [22m │ [1mValue     [22m │
├────────────┼────────────┤
│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders table with text alignment 1`] = `
[
  "┌────────────┬────────────┬────────────┐
│ [1mLeft      [22m │ [1m  Center  [22m │ [1m     Right[22m │
├────────────┼────────────┼────────────┤
│ L          │     C      │          R │
└────────────┴────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders with auto width 1`] = `
[
  "┌───────┬───────────────────────┐
│ [1mShort[22m │ [1mLong                 [22m │
├───────┼───────────────────────┤
│ A     │ This is a longer text │
│ B     │ Short                 │
└───────┴───────────────────────┘
",
]
`;

exports[`table (isCI = true) > renders with content width 1`] = `
[
  "┌──────┬─────────────┐
│ [1mName[22m │ [1mDescription[22m │
├──────┼─────────────┤
│ A    │ First       │
│ BB   │ Second item │
│ CCC  │ Third       │
└──────┴─────────────┘
",
]
`;

exports[`table (isCI = true) > renders with header styling 1`] = `
[
  "┌────────────┬────────────┐
│ [36mName      [39m │ [36mValue     [39m │
├────────────┼────────────┤
│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > renders without header 1`] = `
[
  "│ Test       │ 123        │
└────────────┴────────────┘
",
]
`;

exports[`table (isCI = true) > truncates long text with ellipsis 1`] = `
[
  "┌─────────────────┐
│ [1mText           [22m │
├─────────────────┤
│ This is a ve... │
└─────────────────┘
",
]
`;
